pkg/cri/sbserver/sandbox_portforward_linux.go

```diff
@@ -127,7 +127,9 @@ func (c *criService) portForward(ctx context.Context, id string, port int32, str
 	})
 
 	if err != nil {
-		return fmt.Errorf("failed to execute portforward in network namespace %q: %w", netNSPath, err)
+		// never decorate err, so that the outer layer can clearly know err underlying type
+		// and decide whether to reset the stream.
+		return err
```

Comment:
Shouldn't `errors.Is` handle the unwrapping just fine?

Reply:
It can also be processed here, the current PR code is processed in portforward.httpstream. Do you suggest to deal with it directly here?

Reply:
No, I'm just confused on the comment. I think the only place you could handle things is where you have it right now, but it'd be nice to log the wrapped error we put together here before resetting the stream in httpstream.go

Reply:
I suggest returning the error instead of processing it here, and the caller performs corresponding error processing on the stream according to the raw err

Reply:
The returned error uses `%w` to wrap the original error, and `errors.Is` can determine if it is the original error.

Reply:
The main thing I'm trying to say is the comments worded as if we wouldn't be able to tell the underlying error if it's wrapped, but the errors.Is(err, syscall.EPIPE) should work the same with it. I'd like to keep the context on the error and log it either here or in httpstream.go

Reply:
ok, I am more clear about the specific working principle of fmt.Errorf(%w), thank you two, I have fixed it, PTAL